lightning/src/ln/channelmonitor.rs

```diff
@@ -2010,7 +2010,7 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 								watch_outputs.push(new_outputs);
 							}
 						}
-						claimable_outpoints.push(new_outpoints);
+						claimable_outpoints.append(&mut new_outpoints);
```

Comment:
It was done this way to avoid mixing spend between revoked commitment txn and HTLC-txn in same block. Our tracking stuff should be reliable enough in case of one of them being reorg-out but still, make harder to reason on it...

Reply:
Hmm, right, but it doesn't do that on master, either. the behavior didn't change in this commit, your last refactor before merge changed that.

Reply:
And, also, I think we *should* aggregate such outputs - reorgs are rare-ish, and if we have to fall back to individual transactions to deal with them so be it, but in the common case it saves a good chunk on fees, so we should do it.

Reply:
No my last change was segregating them inside Vec<> and then iterating inside OnChainTxHandler::block_connected ? Let's move forward with this but add an issue somewhere we should add test-coverage for this case

Reply:
They all get added to the same aggregated_claim map, though.